libbeat/outputs/elasticsearch/config_otel_test.go

```diff
@@ -20,69 +20,186 @@ package elasticsearch
 import (
 	_ "embed"
 	"fmt"
-	"os"
-	"path/filepath"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"go.opentelemetry.io/collector/confmap/confmaptest"
+	"go.opentelemetry.io/collector/confmap"
 	"gopkg.in/yaml.v2"
 
 	"github.com/elastic/elastic-agent-libs/config"
 )
 
 func TestToOtelConfig(t *testing.T) {
 
-	tests := []struct {
-		name      string
-		input     string
-		expOutput string
-		experr    bool
-	}{
-		{
-			name:      "basic elasticsearch input",
-			input:     "basic.yml",
-			expOutput: "basic-op.yml",
-			experr:    false,
-		},
-	}
-
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			rawConf, err := loadBeatConf(filepath.Join("testdata", test.input))
-			require.NoError(t, err)
-			beatCfg := config.MustNewConfigFrom(rawConf)
-
-			otelCfg, err := ToOTelConfig(beatCfg)
-			require.NoError(t, err, "could not convert beat config to otel ES config")
-
-			expectedValue, err := confmaptest.LoadConf(filepath.Join("testdata", test.expOutput))
-			require.NoError(t, err)
-			want, err := yaml.Marshal(expectedValue.ToStringMap())
-			require.NoError(t, err)
-
-			got, err := yaml.Marshal(otelCfg)
-			require.NoError(t, err)
-
-			assert.Equal(t, string(want), string(got))
-
-		})
-	}
+	t.Run("basic config translation", func(t *testing.T) {
+		basicInput := `
+hosts:
+  - localhost:9200
+  - localhost:9300
+protocol: http
+path: /foo/bar
+username: elastic
+password: changeme
+index: "some-index"
+pipeline: "some-ingest-pipeline"
+proxy_url: "https://proxy.url"
+backoff:
+  init: 42s
+  max: 420s
+workers: 30
+headers:
+  X-Header-1: foo
+  X-Bar-Header: bar`
+
+		basicOp := `
+api_key: ""
+endpoints:
+  - http://localhost:9200/foo/bar
+  - http://localhost:9300/foo/bar
+idle_conn_timeout: 3s
+logs_index: some-index
+num_workers: 30
+password: changeme
+pipeline: some-ingest-pipeline
+proxy_url: https://proxy.url
+retry:
+  enabled: true
+  initial_interval: 42s
+  max_interval: 7m0s
+  max_retries: 3
+timeout: 1m30s
+user: elastic
+headers:
+  X-Header-1: foo
+  X-Bar-Header: bar    
+batcher:
+  enabled: true
+  max_size_items: 1600
+ `
+		input := newFromYamlString(t, basicInput)
+		cfg := config.MustNewConfigFrom(input.ToStringMap())
+		got, err := ToOTelConfig(cfg)
+		require.NoError(t, err, "error translating elasticsearch output to OTel ES exporter type")
+		expOutput := newFromYamlString(t, basicOp)
+		compareAndAssert(t, expOutput, confmap.NewFromStringMap(got))
+
+	})
+
+	// when preset is configured, we only test worker, bulk_max_size, idle_connection_timeout here
+	// TODO: Check for compression_level when we add support upstream
+	t.Run("check preset config translation", func(t *testing.T) {
+		commonInput := `
+hosts:
+  - localhost:9200
+index: "some-index"
+username: elastic
+password: changeme
+preset: %s
+`
+
+		commonOutput := `
```

Comment:
Same thing here, maybe we can call them `beatCfg` or `beatOutputCfg` and the expected one `otelCfg` or `otelExporterCfg`